github/issues.go

```diff
@@ -131,6 +132,9 @@ func (s *IssuesService) listIssues(u string, opt *IssueListOptions) ([]Issue, *R
 		return nil, nil, err
 	}
 
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeReactionsPreview)
```

Comment:
I don't think this is related to this library, but rather the GitHub API... But I'll mention it here because I don't know of a better place.

It's just an observation. I see that supplying this custom Accept header has a small change on the behavior of `pull_request` field inside the `Issue` struct when it's an issue (not pull request).

Anyway, the change is that with the custom Accept header, that `pull_request` field is no longer omitted from the JSON. Notice:

```diff
~ $ diff -u <(curl 'https://api.github.com/repos/gorilla/mux/issues/2') \
            <(curl 'https://api.github.com/repos/gorilla/mux/issues/2' -H "Accept: application/vnd.github.squirrel-girl-preview")
--- /dev/fd/63	2016-05-14 13:36:15.000000000 -0700
+++ /dev/fd/62	2016-05-14 13:36:15.000000000 -0700
@@ -65,6 +65,22 @@
   "created_at": "2012-10-05T13:25:48Z",
   "updated_at": "2016-03-12T19:50:58Z",
   "closed_at": null,
+  "pull_request": {
+    "url": null,
+    "html_url": null,
+    "diff_url": null,
+    "patch_url": null
+  },
   "body": "Old issue: http://code.google.com/p/gorilla/issues/detail?id=51\r\n\r\nAccording to section 14.23 of RFC 2616 the Host header can include the port number if the default value of 80 is not used.\r\n\r\nSee https://github.com/gorilla/mux/blob/master/mux_test.go#L119 for a commented-out test case that fails.\r\n\r\nPotential fix is editing getHost() https://github.com/gorilla/mux/blob/master/regexp.go#L237 so it has a flag as to whether to strip off the port information. A new field would need to be probably need to be added somewhere such as routeRegexp.\r\n\r\nThe main reason I was using Host() match with a port in the first place is for trying to build reverse routes that include the absolute path as opposed to a relative one.",
-  "closed_by": null
+  "closed_by": null,
+  "reactions": {
+    "url": "https://api.github.com/repos/gorilla/mux/issues/2/reactions",
+    "total_count": 0,
+    "+1": 0,
+    "-1": 0,
+    "laugh": 0,
+    "hooray": 0,
+    "confused": 0,
+    "heart": 0
+  }
 }
```

But it means, with this PR applied, it's no longer to do this to check if an issue is a PR:

```Go
for _, issue := range ghIssuesAndPRs {
	// Filter out PRs.
	if issue.PullRequestLinks != nil {
		continue
	}
	... process issues only
```

Instead, one must now do something like this:

```Go
for _, issue := range ghIssuesAndPRs {
	// Filter out PRs.
	if issue.PullRequestLinks != nil && issue.PullRequestLinks.URL != nil {
		continue
	}
	... process issues only
```

Anyway, I wanted to point this out, because I don't think this change is intended by you @gmlewis. At the same time, it's not really our place to be modifying JSON that GitHub API returns. Maybe this is a common thing to all preview APIs? Or maybe it's an issue we should let them know about? I don't know.